Sources/ArcGISToolkit/Components/AR/TableTopSceneView.swift

```diff
@@ -0,0 +1,249 @@
+    @State private var initialTransformation: TransformationMatrix? = nil
+    
+    private let cameraController: TransformationMatrixCameraController
+    private let sceneViewBuilder: (SceneViewProxy) -> SceneView
+    private let configuration: ARWorldTrackingConfiguration
+    private var initialTransformationIsSet: Bool {
+        initialTransformation != nil
+    }
+    
+    /// Creates a fly over scene view.
+    /// - Parameters:
+    ///   - anchorPoint: The anchor point of the ArcGIS Scene.
+    ///   - translationFactor: The translation factor that defines how much the scene view translates
+    ///   as the device moves.
+    ///   - clippingDistance: Determines the clipping distance in meters around the camera. A value
+    ///   of `nil` means that no data will be clipped.
+    ///   - sceneView: A closure that builds the scene view to be overlayed on top of the
+    ///   augmented reality video feed.
+    /// - Remark: The provided scene view will have certain properties overridden in order to
+    /// be effectively viewed in augmented reality. Properties such as the camera controller,
+    /// and view drawing mode.
+    public init(
+        anchorPoint: Point,
+        translationFactor: Double,
+        clippingDistance: Double?,
+        @ViewBuilder sceneView: @escaping (SceneViewProxy) -> SceneView
+    ) {
+        self.sceneViewBuilder = sceneView
+        
+        let initialCamera = Camera(location: anchorPoint, heading: 0, pitch: 90, roll: 0)
+        cameraController = TransformationMatrixCameraController(originCamera: initialCamera)
+        cameraController.translationFactor = translationFactor
+        cameraController.clippingDistance = clippingDistance
+        
+        configuration = ARWorldTrackingConfiguration()
+        configuration.worldAlignment = .gravityAndHeading
+        configuration.planeDetection = [.horizontal]
+    }
+    
+    public var body: some View {
+        ZStack {
+            ARSwiftUIView(proxy: arViewProxy)
+                .onRender { _, _, _ in
+                    guard let sceneViewProxy else { return }
+                    updateLastGoodDeviceOrientation()
+                    sceneViewProxy.draw(
+                        for: arViewProxy,
+                        cameraController: cameraController,
+                        orientation: lastGoodDeviceOrientation,
+                        initialTransformation: initialTransformation ?? .identity
+                    )
+                }
+                .onAddNode { _, node, anchor in
+                    addPlane(with: node, for: anchor)
+                }
+                .onUpdateNode { _, node, anchor in
+                    updatePlane(with: node, for: anchor)
+                }
+                .onAppear {
+                    arViewProxy.session?.run(configuration)
+                }
+                .onDisappear {
+                    arViewProxy.session?.pause()
+                }
+                .gesture(
```

Comment:
shouldn't this just be a tap gesture?

```swift

                .onTapGesture {
                    ...
                }
```

Reply:
`onTapGesture()` does not provide the screen point in iOS 15 and the screen point is needed for hit testing.

Reply:
So I wonder if we write our own onTap that does provide the screen point and internally use the `if available` and if not iOS 16 then in that place use the DragGesture. Then eventually the workaround can be removed. [I did that here](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/v.next/Sources/ArcGISToolkit/Extensions/SwiftUI/View.swift#L62-L71) for presentation detents.

Reply:
I added a `onSingleTapGesture()` modifier similar to the one on `MapView`/`SceneView` which uses UIKit to detect the single tap rather than SwiftUI.